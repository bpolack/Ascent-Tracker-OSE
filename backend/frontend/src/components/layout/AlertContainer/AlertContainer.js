import React, { Component } from 'react'; 
import PropTypes from 'prop-types'; 
import { connect } from 'react-redux'; 

import Alert from './Alert/Alert';

export class AlertContainer extends Component {

	render() {

		const { alerts } = this.props;

		if (alerts && alerts.length > 0) {
			return (
				<div className="fixed z-50 bottom-5 right-5 w-full max-w-xs max-h-64">
					{alerts.map(alert => <Alert key={alert.id} alertId={alert.id} alertType={alert.alertType} msg={alert.msg} />)}
				</div>
			)
		}
		else {
			return (
				<div className="fixed bottom-4 right-4 w-full max-w-xs max-h-64"></div>
			)
		}

	}
}

AlertContainer.propTypes = {
	alerts: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
	alerts: state.alert
});

export default connect(mapStateToProps)(AlertContainer)
