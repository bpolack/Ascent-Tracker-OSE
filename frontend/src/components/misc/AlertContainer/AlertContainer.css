.AlertContainer {
	position: fixed;
	max-width: 350px;
	bottom: 60px;
	right: 40px;
}