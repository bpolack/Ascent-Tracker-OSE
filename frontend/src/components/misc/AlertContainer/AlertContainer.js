import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './AlertContainer.css'; 

import Alert from './Alert/Alert';

export class AlertContainer extends Component {

	render() {

		const { alerts } = this.props;

		if (alerts && alerts.length > 0) {
			return (
				<div className="AlertContainer">
					{alerts.map(alert => <Alert key={alert.id} alertId={alert.id} alertType={alert.alertType} msg={alert.msg} />)}
				</div>
			)
		}
		else {
			return (
				<div className="AlertContainer"></div>
			)
		}

	}
}

AlertContainer.propTypes = {
	alerts: PropTypes.array.isRequired
}

const mapStateToProps = state => ({
	alerts: state.alert
});

export default connect(mapStateToProps)(AlertContainer)
