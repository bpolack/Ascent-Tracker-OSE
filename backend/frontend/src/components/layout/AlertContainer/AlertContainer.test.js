import React from 'react';
import { shallow } from 'enzyme';
import { AlertContainer } from './AlertContainer';

describe('AlertContainer', () => {
	it("Renders correctly without crashing", () => {
		const alerts = [];
		const wrapper = shallow(<AlertContainer alerts={alerts} />);
		expect(wrapper).toMatchSnapshot();
	});
});