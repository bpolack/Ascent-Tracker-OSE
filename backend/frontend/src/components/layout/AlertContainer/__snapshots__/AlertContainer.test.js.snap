// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertContainer Renders correctly without crashing 1`] = `ShallowWrapper {}`;
